From e11dafd3691a646332f2c44317b08ff82ee93c52 Mon Sep 17 00:00:00 2001
From: Leonid Komarianskyi <leonid_komarianskyi@epam.com>
Date: Mon, 27 May 2024 10:08:29 +0300
Subject: [PATCH 1/4] gen5: vdk: workaround: fix address usage

Previous address usage leads to SError in Xen when traps are enabled.
Changed this value according to simulation.

Tested on VDK 3.1.0.

Signed-off-by: Leonid Komarianskyi <leonid_komarianskyi@epam.com>
---
 arch/arm/mach-rmobile/cpu_info-rcar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rmobile/cpu_info-rcar.c b/arch/arm/mach-rmobile/cpu_info-rcar.c
index d467a100fc..192110d3fd 100644
--- a/arch/arm/mach-rmobile/cpu_info-rcar.c
+++ b/arch/arm/mach-rmobile/cpu_info-rcar.c
@@ -16,7 +16,7 @@
 static u32 rmobile_get_prr(void)
 {
 	if (IS_ENABLED(CONFIG_RCAR_64))
-		return readl(0xFFF00044);
+		return readl(0x187D0044);
 }
 
 #define MIDR_PARTNUM_CORTEX_A720	0xD81
-- 
2.25.1

